src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2437,7 +2445,8 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                 VisitRvalue(binary.Right);
                 Debug.Assert(!IsConditionalState);
-                Debug.Assert(this.State.Reachable);
+                // At this point, State.Reachable may be false for
```

Comment:
>// At this point, State.Reachable may be false for [](start = 16, length = 50)

Should we skip the nullability analysis if the state is unreachable? Is it possible that we will corrupt the state in some way?